src/nxt_port_socket.c

```diff
@@ -609,9 +612,29 @@ nxt_port_read_handler(nxt_task_t *task, void *obj, void *data)
         iov[1].iov_base = b->mem.pos;
         iov[1].iov_len = port->max_size;
 
-        n = nxt_socketpair_recv(&port->socket, &msg.fd, iov, 2);
+        n = nxt_socketpair_recv(&port->socket, iov, 2, oob, &oobn);
 
-        if (n > 0) {
+        if (nxt_fast_path(n > 0)) {
+            msg.pid = -1;
+            msg.fd  = -1;
+
+            /**
+             * keeping in one branch because the slow path can only
+             * ever happen if client is trying to spoof creds on
+             * platforms that cred is enabled (linux and freebsd).
+             */
+            if (nxt_slow_path(
+                    oobn > 0 &&
+                    nxt_socket_msg_oob_info(oob, oobn, &msg.fd, &msg.pid)
+                        != NXT_OK)) {
+                nxt_alert(task, "failed to get oob data from %d", port->socket);
+                goto fail;
+            }
+
+            if (nxt_slow_path(msg.pid == -1)) {
```

Comment:
This block should be removed after conditional initialisation of `msg.pid` above will be done.